runtime/src/main/java/org/corfudb/runtime/collections/CorfuTable.java

```diff
@@ -658,6 +667,9 @@ protected void unmapSecondaryIndexes(K key, V value) {
                     Map<K, V> slot = secondaryIndex.get(indexKey);
                     if (slot != null) {
                         slot.remove(key, value);
+                        if (slot.isEmpty()) {
```

Comment:
when can this happen?

Whenever the slot is created, a key-value pair is always inserted -
https://github.com/CorfuDB/CorfuDB/blob/master/runtime/src/main/java/org/corfudb/runtime/collections/CorfuTable.java#L697

Unless I am missing something...

Reply:
When a key/value pair is instered into the table, multiple values can be mapped in the secondary index. When the key is removed from the primary map, the unmapSecondaryIndexes doesn't remove the Map container created in https://github.com/CorfuDB/CorfuDB/blob/master/runtime/src/main/java/org/corfudb/runtime/collections/CorfuTable.java#L696 when it's empty causing a memory leak. 

You can revert my fix and run the unit test in a debugger and can inspect the internal map to see that it's not removed, but its empty.